src/sdf_assets.rs

```diff
@@ -15,26 +15,39 @@ pub trait SdfAssets {
 impl SdfAssets for Assets<Shader> {
     fn add_sdf_body<T: Into<String>>(&mut self, sdf: T) -> Handle<Shader> {
         let body = sdf.into();
-        let shader = Shader::from_wgsl(format!(
-            r#"
-#import bevy_smud::shapes
+        let id = Uuid::new_v4();
+        let shader = Shader::from_wgsl(
+            format!(
+                r#"
+#define_import_path bevy_smud::sdf{id}
+
+#import bevy_smud::shapes as shapes
+
 fn sdf(p: vec2<f32>) -> f32 {{
     {body}
 }}
 "#
-        ));
+            ),
+            file!(),
+        );
         self.add(shader)
     }
 
     fn add_fill_body<T: Into<String>>(&mut self, fill: T) -> Handle<Shader> {
         let body = fill.into();
-        let shader = Shader::from_wgsl(format!(
-            r#"
+        let id = Uuid::new_v4();
+        let shader = Shader::from_wgsl(
+            format!(
+                r#"
+#define_import_path bevy_smud::fill{id}
+
 fn fill(d: f32, color: vec4<f32>) -> vec4<f32> {{
     {body}
 }}
 "#
-        ));
+            ),
+            file!(),
```

Comment:
What's this about?

Reply:
AFAIK this field for debugging purposes, to see "filename" in logs. file! will name it `sdf_assets.rs` but I think it's better just put `bevy_smud::fill` instead.